kubernetes-client/src/com/goyeau/kubernetes/client/operation/Replaceable.scala

```diff
@@ -13,12 +13,11 @@ import org.http4s.Method._
 
 private[client] trait Replaceable[F[_], Resource <: { def metadata: Option[ObjectMeta] }] {
   protected def httpClient: Client[F]
-  implicit protected val F: Sync[F]
   protected def config: KubeConfig
   protected def resourceUri: Uri
   implicit protected def resourceEncoder: Encoder[Resource]
 
-  def replace(resource: Resource): F[Status] =
+  def replace(resource: Resource)(implicit F: Sync[F]): F[Status] =
```

Comment:
I actually don't think this is a good idea to move the `Sync[F]` to function level because my initial design was anticipating that from Scala 3 we will have the ability to pass implicit parameters on traits. So those `protected def` will move to trait parameters on Scala 3.

Reply:
If its at the method level, you can add the requirements what the method actually needs, like if you are not doing any decoding, you dont need a Sync[F] instance, but maybe just Monad[F], or MonadError[F, Throwable]

Reply:
I would normally agree with this statement but here the intent is a little different: We are applying bounds to what capabilities `F` needs to have in order to fulfill it's purpose ("replace" stuff).

With this change you give the ability to create a `Replaceable` client with a `F` that potentially does not provide `Sync` capabilities and then the user would not be able to call `replace`, which is good but the trait kind of broke it's promise of providing ability to "replace" stuff.

Again I agree that the current mixing/overriding `F` is cumbersome but that will be greatly improved with [Scala 3 trait parameters](http://dotty.epfl.ch/docs/reference/other-new-features/trait-parameters.html).

Reply:
I understand the intent. You will still need an IO type, since you cannot create a Client without it.

Reply:
I expect this code would look like this in Scala 3:
```
private[client] trait Replaceable[F[_]: Sync, Resource <: { def metadata: Option[ObjectMeta] }: Encoder](config: KubeConfig, resourceUri: Uri) {
  def replace(resource: Resource)(implicit F: Sync[F]): F[Status] = ???
}
```

Reply:
CE3 will change also what is needed. 
JsonEncoders do not really need a Sync, and if we start using JsonDecoder, we dont need Sync there either.
I can see if I can come up with an encoding for what I mean.

Reply:
references: 
https://github.com/http4s/http4s/blob/main/circe/src/main/scala/org/http4s/circe/CirceEntityEncoder.scala
and https://github.com/http4s/http4s/blob/main/circe/src/main/scala/org/http4s/circe/JsonDecoder.scala

Reply:
if you prefer I can do that as its own thing, and remove the sync changes

Reply:
That sounds good @hamnis